### PR TITLE
Teach CPU detection routine about OpenPOWER

### PR DESCRIPTION
**Pull request by madscientist159**
Tested on Talos II and in pSeries virtual machine

### CODE REVIEW

**Comment by kesselb** [comment]
Thanks for your pull request :+1: 

Please add a test to https://github.com/nextcloud/serverinfo/blob/master/tests/lib/DefaultOsTest.php

**Comment by madscientist159** [comment]
Anything else I need to do before this can be merged?

**Comment by kesselb** [comment]
Thank you! The test looks fine. 

I realized we don't have a test for the Raspberry Pi cpu name code and added one: https://github.com/nextcloud/serverinfo/pull/440

Please rebase your branch to pull the changes in.

**Comment by kesselb** [comment]
Hi, 

the current way to extract the cpu name was a bit complicated. I made another pr to improve it: https://github.com/nextcloud/serverinfo/pull/441

It should be much easier now to integrate the OpenPower changes. 
Something like:

```
elseif (str_contains($cpuinfo, 'PowerNV') || str_contains($cpuinfo, 'CHRP IBM pSeries')) {
			$pattern = '/cpu\s+:\s+(.+)/';
		} 
```

File: tests/lib/DefaultOsTest.php [hunk 1]
```diff
@@ -117,6 +117,14 @@ public function testGetCPUNamePi4b(): void {
 		$this->assertEquals('Raspberry Pi 4 Model B Rev 1.2 (4 cores)', $this->os->getCpuName());
 	}
 
+	public function testGetCPUNameOpenPower(): void {
```
Comment (kesselb) [inline tests/lib/DefaultOsTest.php:120]:
```suggestion
	public function testGetCpuNameOpenPower(): void {
```

**Review by kesselb — APPROVED** [review]

**Pull request merged**